src/Microsoft.Health.Fhir.Shared.Core/Features/Resources/Delete/DeletionService.cs

```diff
@@ -192,7 +193,24 @@ public async Task<long> DeleteMultipleAsync(ConditionalDeleteResourceRequest req
                 await cancellationTokenSource.CancelAsync();
             }
 
-            System.Threading.Tasks.Task.WaitAll(deleteTasks.ToArray(), cancellationToken);
+            try
+            {
+                // We need to wait until all running tasks are cancelled to get a count of resources deleted.
+                Task.WaitAll(deleteTasks.ToArray(), cancellationToken);
+            }
+            catch (AggregateException age)
+            {
+                // If one of the tasks fails, the rest may throw a cancellation exception. Filtering those out as they are noise.
+                foreach (var coreException in age.InnerExceptions.Where(e => e is not TaskCanceledException))
```

Comment:
Should TaskCancelledExceptions also be filtered out on line 223 when we make the exception to add to the job record?

Reply:
I think we still want 223 (now 224) to count resources deleted on threads that have been cancelled??

I filtered out the `TaskCancelled` exceptions from the thrown `AggregateException`. Thoughts about this?

Reply:
Yes, you what I was thinking of but didn't explain correctly.